src/context/WalletProvider/Ledger/components/Chains.tsx

```diff
@@ -0,0 +1,161 @@
+import { Box, Button, Flex, ModalBody, Spinner, Text as CText } from '@chakra-ui/react'
+import type { ChainId } from '@shapeshiftoss/caip'
+import {
+  bchAssetId,
+  bchChainId,
+  btcAssetId,
+  btcChainId,
+  dogeAssetId,
+  dogeChainId,
+  ethAssetId,
+  ethChainId,
+  fromAccountId,
+  fromChainId,
+  ltcAssetId,
+  ltcChainId,
+} from '@shapeshiftoss/caip'
+import pull from 'lodash/pull'
+import { useCallback, useMemo, useState } from 'react'
+import { AssetIcon } from 'components/AssetIcon'
+import { Text } from 'components/Text'
+import { getSupportedEvmChainIds } from 'hooks/useEvm/useEvm'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { deriveAccountIdsAndMetadataForChainNamespace } from 'lib/account/account'
+import type { BN } from 'lib/bignumber/bignumber'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { isSome } from 'lib/utils'
+import { portfolio, portfolioApi } from 'state/slices/portfolioSlice/portfolioSlice'
+import { selectAssets, selectWalletChainIds } from 'state/slices/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+export const LedgerChains = () => {
+  const { state: walletState } = useWallet()
+  const dispatch = useAppDispatch()
+  const assets = useAppSelector(selectAssets)
+
+  const walletChainIds = useAppSelector(selectWalletChainIds)
+
+  const availableAssetIds = useMemo(
+    () => [btcAssetId, dogeAssetId, bchAssetId, ltcAssetId, ethAssetId],
+    [],
+  )
+
+  const availableAssets = useMemo(
+    () => availableAssetIds.map(id => assets[id]).filter(isSome),
+    [assets, availableAssetIds],
+  )
+
+  const availableChainIds = useMemo(
+    () => [btcChainId, dogeChainId, bchChainId, ltcChainId, ethChainId],
+    [],
+  )
+
+  const [loadingChains, setLoadingChains] = useState<Record<ChainId, boolean>>({})
+
+  const handleConnectClick = useCallback(
+    async (chainId: ChainId) => {
+      if (!walletState.adapters) return
+      if (!walletState?.wallet) return
+
+      setLoadingChains(prevLoading => ({ ...prevLoading, [chainId]: true }))
+
+      try {
+        const { chainNamespace } = fromChainId(chainId)
+        const chainIds = chainId === ethChainId ? getSupportedEvmChainIds() : [chainId]
+        const accountMetadataByAccountId = await deriveAccountIdsAndMetadataForChainNamespace[
+          chainNamespace
+        ]({
+          accountNumber: 0,
+          chainIds,
+          wallet: walletState.wallet,
+        })
+
+        const accountIds = Object.keys(accountMetadataByAccountId)
+        const { getAccount } = portfolioApi.endpoints
+        const opts = { forceRefetch: true }
+        const accountPromises = accountIds.map(accountId =>
+          dispatch(getAccount.initiate({ accountId }, opts)),
+        )
+        const accountResults = await Promise.allSettled(accountPromises)
+
+        const balanceByChainId = accountResults.reduce<Record<ChainId, BN>>((acc, res, idx) => {
+          if (res.status === 'rejected') return acc
```

Comment:
do we want logging for failed requests here?

Reply:
We absolutely may - 9f7a6191